packages/api/src/utils/logs.js

```diff
@@ -171,7 +178,7 @@ export class Logging {
       await this.postBatch()
     }
     this.ctx.waitUntil(run())
-
+    this._finished = true
```

Comment:
minor, but let's move this up immediately after the `if (this._finished)` check.

Reply:
Good point. Done.